libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/PhoneState.kt

```diff
@@ -11,19 +11,107 @@ import com.mapbox.android.telemetry.TelemetryUtils.retrieveVendorId
 import com.mapbox.navigation.core.telemetry.obtainAudioType
 import com.mapbox.navigation.core.telemetry.obtainScreenBrightness
 import com.mapbox.navigation.core.telemetry.obtainVolumeLevel
+import org.jetbrains.annotations.TestOnly
 
 /**
  * Class that will hold the current states of the device phone.
  */
-internal data class PhoneState(val context: Context) {
-    val volumeLevel: Int = obtainVolumeLevel(context)
-    val batteryLevel: Int = obtainBatteryLevel(context)
-    val screenBrightness: Int = obtainScreenBrightness(context)
-    val isBatteryPluggedIn: Boolean = isPluggedIn(context)
-    val connectivity: String = obtainCellularNetworkType(context)
-    val audioType: String = obtainAudioType(context)
-    val applicationState: String = obtainApplicationState(context)
-    val created: String = obtainCurrentDate()
-    val feedbackId: String = obtainUniversalUniqueIdentifier()
-    val userId: String = retrieveVendorId()
+internal class PhoneState {
```

Comment:
Why transform from `data class` to regular `class`?

`data class` could have default values for all arguments (besides `context`) while in tests we could swap them out without having to maintain `equals/hashcode/toString` manually.

Ideally, though, we should rather mock the providers since those are separate methods now anyway.

Reply:
it needed to avoid storing `context` as a `val` because this class should be serialized-deserialized. Changed to data and added `companion` to create a new instance statically

Reply:
I believe that if you would provide context as `data class Clazz(context: Context)` instead of `data class Clazz(private val context: Context)`, the `context` wouldn't be serialized, it's only in the constructor scope and then dropped.

Reply:
you cannot have non-`val/var` parameters in the `data` constructor. Anyway, I've already changed it, new approach avoids using context in the constructor